core/src/test/java/kafka/test/junit/DetectThreadLeakTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.test.junit;
+
+import org.apache.kafka.test.TestUtils;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DetectThreadLeakTest {
```

Comment:
This is a test that is testing other tests and IMHO is not necessary. You should at least call join() to ensure the thread is not leaked from this test :^)

Reply:
> This is a test that is testing other tests and IMHO is not necessary.

We need a test to make sure DetectThreadLeak works well. For example, it becomes a no-op due to a bug, so everything work well. include thread leak :smile: 

> You should at least call join() to ensure the thread is not leaked from this test :^)

totally agree